src/sas/qtgui/Utilities/Preferences/PreferencesWidget.py

```diff
@@ -142,6 +148,31 @@ def addIntegerInput(self, title: str, default_number: Optional[int] = 0) -> QLin
         int_box.setValidator(QIntValidator())
         return int_box
 
+    def _validate_input_and_stage(self, edit: QLineEdit, key: str):
+        """A generic method to validate values entered into QLineEdit inputs. If the value is acceptable, it is staged,
+        otherwise, the input background color is changed to yellow and any previous changes will be unstaged until the
+        value is corrected.
+        :param edit: The QLineEdit input that is being validated.
+        :param key: The string representation of the key the QLineEdit value is stored as in the configuration system.
+        :return: None
+        """
+        edit.setStyleSheet("background-color: white")
+        validator = edit.validator()
+        text = edit.text()
+        (state, val, pos) = validator.validate(text, 0) if validator else (0, 0, 0)
+        if state == QValidator.Acceptable:
+            if isinstance(validator, QIntValidator):
+                self._stageChange(key, int(text))
+            elif isinstance(validator, QDoubleValidator):
+                self._stageChange(key, float(text))
+            else:
+                self._stageChange(key, text)
```

Comment:
This bothers me a bit. Why should _stageChange need to know the type? Can't we just stage a string representation of the value? Then, at the unstaging step, the widget itself (GPU Options, Display Settings etc) determines what to cast this into. 
By no means is this comment related to this PR, it's more of a discussion point.